migrate to github actions

HarrisonWAffel opened the pull request:
This PR migrates CI away from drone towards github actions. 

A successful release CI run can be seen here: https://github.com/HarrisonWAffel/windows_exporter-package/actions/runs/9087280411
The resulting docker images can be found here: https://hub.docker.com/layers/harrisonwaffel/windows-exporter-package/v0.0.0-rc2/images/sha256-1e97a6589140e15cc44573cbbc0124a9bf7b923ae8fe67a192cb5a94f700650e?context=explore

HarrisonWAffel committed:
migrate to gha

jiaqiluo commented on .github/workflows/release.yaml:
```diff
@@ -0,0 +1,91 @@
+      ARCH: amd64
+      OS_VERSION: ltsc2022
+    steps:
+      - uses: actions/checkout@v1
```
```suggestion
      - uses: actions/checkout@v4
```

jiaqiluo commented on .github/workflows/release.yaml:
```diff
@@ -0,0 +1,91 @@
+      ARCH: amd64
+      OS_VERSION: ltsc2019
+    steps:
+      - uses: actions/checkout@v1
```
```suggestion
      - uses: actions/checkout@v4
```

jiaqiluo commented on .github/workflows/release.yaml:
```diff
@@ -0,0 +1,91 @@
+          docker push ${{ env.REPO }}/windows-exporter-package:${{ env.TAG }}-windows-${{ env.OS_VERSION }}
+
+  create-manifest:
+    needs: [ build-push-2019-image ]
```
Should it need both build jobs?

HarrisonWAffel replied:
i did it this way to limit the number of runners that are spawned at once, since the rancher org has a limit of 20 at a given time. 2019 depends on 2022, so `create-manifest` will only run when both of those steps are complete

jiaqiluo commented on .github/workflows/release.yaml:
```diff
@@ -0,0 +1,91 @@
+          password: ${{ env.DOCKER_PASSWORD }}
+      - name: create and push manifest
+        run: |
+          docker manifest create ${{ env.REPO }}/windows-exporter-package:${{ env.TAG }} ${{ env.REPO }}/windows-exporter-package:${{ env.TAG }}-windows-ltsc2022 ${{ env.REPO }}/windows-exporter-package:${{ env.TAG }}-windows-ltsc2019
```
Old "child" tags have the suffix `-1089` and `-ltsc2022`, but I think the change is ok because the "parent" tag is the same and should be used.

HarrisonWAffel replied:
Yeah this was conscious effort to standardize the actual suffix's with the version names that are commonly used online. 1809 and 2019 are the same, but ltsc 2019 is more clear since that's the same of the actual windows server version used by customers

HarrisonWAffel commented:
@jiaqiluo I've addressed your comments. 

Also, to give some additional context around this repository: Once 2.8 is no longer supported this repository will be archived https://github.com/rancher/windows/issues/231, so there's a very low chance these workflows will actually ever see use
